packages/editor-ui/src/components/ValueSurvey.vue

```diff
@@ -0,0 +1,189 @@
+<template>
+	<ModalDrawer
+		:name="VALUE_SURVEY_MODAL_KEY"
+		:beforeClose="closeDialog"
+		:modal="false"
+		:wrapperClosable="false"
+		direction="btt"
+		width="120px"
+		class="value-survey"
+	>
+		<template slot="header">
+			<span :class="$style.title" v-html="getTitle" />
+		</template>
+		<template slot="content">
+			<section :class="$style.content">
+				<div v-if="showButtons" :class="$style.buttons">
+					<n8n-button @click="selectSurveyValue('very')" label="Very disappointed" />
+					<n8n-button @click="selectSurveyValue('somewhat')" label="Somewhat disappointed" />
+					<n8n-button @click="selectSurveyValue('not')" label="Not disappointed (it isn’t really that useful)" />
+				</div>
+				<div v-else :class="$style.email">
+					<div :class="$style.input">
+						<n8n-input
+							v-model="form.email"
+							placeholder="Your email address"
+						/>
+						<n8n-button label="Send" float="right" @click="send" :disabled="!isEmailValid" />
+					</div>
+					<div :class="$style.disclaimer">David from our product team will get in touch personally.</div>
+				</div>
+			</section>
+		</template>
+	</ModalDrawer>
+</template>
+
+<script lang="ts">
+import mixins from "vue-typed-mixins";
+
+import { VALID_EMAIL_REGEX, VALUE_SURVEY_MODAL_KEY } from '@/constants';
+import { workflowHelpers } from "@/components/mixins/workflowHelpers";
+import ModalDrawer from './ModalDrawer.vue';
+
+const DEFAULT_TITLE = `How would you feel if you could <strong>no longer use n8n</strong>?`;
+const VERY_TITLE = `Great to hear! Can we reach out to see how we can make n8n even better for you?`;
+const SOMEWHAT_TITLE = `Thanks for your feedback! We'd love to understand how we can improve. Can we reach out?`;
+const NOT_TITLE = `Sorry to hear that. We'd love to learn how to improve. Can we reach out?`;
+
+export default mixins(workflowHelpers).extend({
+	name: 'ValueSurvey',
+	components: {
+		ModalDrawer,
+	},
+	computed: {
+		getTitle (): string {
+			if (this.form.value === 'very') {
+				return VERY_TITLE;
+			} else if (this.form.value === 'somewhat') {
+				return SOMEWHAT_TITLE;
+			} else if (this.form.value === 'not') {
+				return NOT_TITLE;
+			} else {
+				return DEFAULT_TITLE;
+			}
+		},
+		isEmailValid(): boolean {
+			return VALID_EMAIL_REGEX.test(String(this.form.email).toLowerCase());
+		},
+	},
+	data() {
+		return {
+			form: {
+				email: '',
+				value: '',
+			},
+			showButtons: true,
+			VALUE_SURVEY_MODAL_KEY,
+		};
+	},
+	methods: {
+		closeDialog(): void {
+			if (this.form.value === '') {
+				this.$telemetry.track('User responded value survey score', { instance_id: this.$store.getters.instanceId, how_disappointed: '' });
+			} else if (this.form.value !== '') {
+				this.$telemetry.track('User responded value survey email', { instance_id: this.$store.getters.instanceId, email: '' });
+			}
+
+			this.$store.commit('ui/closeTopModal');
```

Comment:
this is called twice on `beforeClose`. this will close the top modal even if it's not current modal since it's also called from within the modal on `close`..
I would remove this and simply call `this.modalBus.$emit('close')`  after sending data

Reply:
same thing applies to other modal

Reply:
It does not work that way :/